src/python/pants/backend/python/dependency_inference/import_parser.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import ast as ast3
+import re
+import sys
+import warnings
+from dataclasses import dataclass
+from typing import Set
+
+from typed_ast import ast27
+
+from pants.util.ordered_set import FrozenOrderedSet
+from pants.util.strutil import ensure_text
+
+
+class ImportParseError(ValueError):
+    pass
+
+
+@dataclass(frozen=True)
+class PythonImports:
+    imported: FrozenOrderedSet[str]
+    inferred: FrozenOrderedSet[str]
```

Comment:
Unclear what these mean. Docs (or clearer names) would be helpful.

Ditto for the classname itself, now that I think of it. What imports is this describing?